### DISCUSSION TITLE
Re-export BaseException for sake of custom API Endpoints

### DISCUSSION THREAD

**Discussion by viters** [directus/directus#10178] [Archive]
Currently `exceptions` passed to `registerEndpoint` are:

```js
{
  ForbiddenException: [Getter],
  GraphQLValidationException: [Getter],
  HitRateLimitException: [Getter],
  IllegalAssetTransformation: [Getter],
  InvalidConfigException: [Getter],
  InvalidCredentialsException: [Getter],
  InvalidIPException: [Getter],
  InvalidOTPException: [Getter],
  InvalidPayloadException: [Getter],
  InvalidQueryException: [Getter],
  MethodNotAllowedException: [Getter],
  RangeNotSatisfiableException: [Getter],
  RouteNotFoundException: [Getter],
  ServiceUnavailableException: [Getter],
  UnprocessableEntityException: [Getter],
  UserSuspendedException: [Getter]
}
```

If none of them match my requirements during implementation of API extensions, I must create an error of instance of BaseException, because of error handler:

```ts
if (errors.some((err) => err instanceof BaseException === false)) {
	res.status(500);
}
```

Right now to do this, you must get prototype of any existing exception:

```js
module.exports = function registerEndpoint(router, { database, exceptions }) {
  const BaseException = Object.getPrototypeOf(exceptions.ForbiddenException);

  class BadRequest extends BaseException {
    constructor(message) {
      super(message, 400, 'BAD_REQUEST');
    }
  }
// ...
```

I suggest re-exporting `BaseException` with `exceptions` for sake of such cases.

**Comment by joselcvarela** [comment]
Hello @viters 
I don't know if we would like to re-export it again, because as you can see here https://github.com/directus/directus/pull/7166/commits/b674b5f4c9be1abb523621a972cee75c6df527cb it were moved into `@directus/shared`.
So my suggestion is to `npm install @directus/shared` and then `import { BaseException } from '@directus/shared/exceptions';`  in your code.

**Reply by rijkvanzanten** [reply]
@joselcvarela I actually just added support for this in https://github.com/directus/directus/pull/10281 to solve for a different discussion (https://github.com/directus/directus/discussions/10276#discussioncomment-1745304) 🙂

**Reply by viters** [reply]
@joselcvarela 
![image](https://user-images.githubusercontent.com/13100466/144941140-7c330779-31a4-46bf-b22e-415ef06a5993.png)
The description of `@directus/shared` discourages to do so.

@rijkvanzanten 
Thanks, that solves the case.

**Comment by viters** [comment]
Can be closed after #10281.
